### ISSUE TITLE
Potential refactoring

### ISSUE THREAD

**Issue by ajmaddaford** [ONSdigital/sdc-cryptography#4]
This is looking good 👍 

Some things I think we should consider refactoring:
- Making the SecretStore a KeyStore and removing functionality that deals with secrets. Secrets can be handled outside of this library via something like https://github.com/theskumar/python-dotenv
- Adding a script that can take a folder of keys and generate a keys yaml file. This would reduce the effort of everyone writing a script to generate a keys file. It should be possible to run this script from a `bin` folder in the Python package. The naming scheme for the files could be something like: 
  ```
  <platform>-<service>-<purpose>-<key_use>-<key_type>-<version>.pem

  platform | Platform the keys are used on e.g. sdc 
  service  | Service the key originates from (holder of the private key) e.g. rrm or eq 
  purpose  | Purpose of the key in the service e.g. authentication or signing
  key_use  | encryption or signing
  key_type | public or private
  version  | Version identifier e.g. v1
  
  e.g. sdc-rrm-authentication-encryption-public-v1.pem
  ```
- Splitting token_helper.py into JWE class with encrypt and decrypt methods and a JWS class with encode and decode methods. This would mean splitting the tests so we clearly define the tests for JWE and JWS seperately.
- Review why we're using the JWT class from jwcrypto rather than the JWS class when dealing with a JWS
- The tests in test_token_helper need to be reviewed:
  - Some tests use the Encoder, others use pre-canned JWT but it's not clear why
  - For the pre-canned JWTs it's not understood how they were generated (and therefore their content)

**Comment by ajmaddaford** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [(5) Create standalone crypto library - python](https://trello.com/c/FzJyUaXM/209-5-create-standalone-crypto-library-python)

**Comment by warrenbailey** [comment]
https://trello.com/c/lVU4s8kn/513-sdc-cryptography-improvements

**Comment by collisdigital** [comment]
Agreed SDX will pick these up in the next Sprint (planning tomorrow) prior to EQ integrating the library @ajmaddaford

**Comment by warrenbailey** [comment]
@ajmaddaford The token_helper class is calling make_token_signed on the JWT class which essentially converts it to a JWS by doing the following:

`t = JWS(self.claims)`
`t.add_signature(key, protected=self.header)`
`self.token = t`

To be honest, it wasn't immediately obvious what it was doing so I am still tempted to change it to using the JWS class directly

**Comment by warrenbailey** [comment]
Right I've tried changing it the JWS class, the encoding works well. It's pretty much exactly the same as the JWT class.

However deserialising with the JWS class and it lacks all the checking of mandatory claims, so I'd have to deserialising into the JWT class as well as the JWS class. 

Looking at it now, it's more complex to do it that way, so I'm going to leave it as is.

**Comment by warrenbailey** [comment]
As for the tests, they look like the ones I wrote  back when I did the original implementation back at the beginning of EQ. If I remember rightly the pre-canned ones were created using the Java nimbus library. At the time I wanted to prove my code work was working correctly so running it against an well tested library was useful. I think they're valuable so i'd be very reluctant to remove them.

What I can't seem to find is the test that used the example from the back of the JWE. That was very useful as that proved the original code work and there's no reason why we couldn't use it against the library.

**Comment by warrenbailey** [comment]
Ah seems it was lost during the EQ PR to move to multiple keys:

https://github.com/ONSdigital/eq-survey-runner/pull/1162/files#diff-bbbbbcc77b7831bf8f8d9d6d42e1d133

Very tempted to recover that test

**Comment by warrenbailey** [comment]
@ajmaddaford is this still valid? 

In the middle of a github tidy up and this is still assigned to me
